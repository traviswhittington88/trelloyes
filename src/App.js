import React from 'react';
import List from './List.js';
import STORE from './store.js';
import './index.css';
import './App.css';
//import { all } from 'q';

function App() {

const cardIdsArray = STORE.lists.map(item => item.cardIds)
const allCardsArray = Object.values(STORE.allCards);
const cards = [];

for(let i = 0; i < allCardsArray.length; i++){
  cards.push(allCardsArray[i]);  
}

for(let j = 0; j < cardIdsArray.length; j++) {
  cards.push(cardIdsArray);
}

const ListsComponentArray = STORE.lists.map((item) => 
  <List header={item.header} key={item.id} cardsArray={item.cardIds} cardsObj={allCardsArray}/>  
)

  return (
    <main className="App">
      <header className="App-header">
        <h1>
          Trelloyes!
        </h1>
      </header>
      <div className="App-list">
        {ListsComponentArray}
      </div>
    </main>
  );

}

export default App;
