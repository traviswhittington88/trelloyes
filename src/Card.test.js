import React, { Component }from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';
import Card from './Card';
import { exportAllDeclaration } from '@babel/types';

describe('Card component', () => {
    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    //snap shot
    it('renders to UI as expected', () => {
        const tree = renderer
            .create(<Card title={'string'} content={'string'} key={'string'}/>)
            .toJSON();
            expect(tree).toMatchSnapshot();
    });
})

