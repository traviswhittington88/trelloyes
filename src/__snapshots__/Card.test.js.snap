// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders to UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    string
  </h3>
  <p>
    string
  </p>
</div>
`;
